Source/MQTTnet/Extensions/UserPropertyExtension.cs

```diff
@@ -0,0 +1,18 @@
+using System;
+using System.Linq;
+
+namespace MQTTnet.Extensions
+{
+    public static class UserPropertyExtension
+    {
+        public static string GetUserProperty(this MqttApplicationMessage message, string propertyName, StringComparison comparisonType = StringComparison.OrdinalIgnoreCase)
+        {
+            return message.UserProperties.SingleOrDefault(up => up.Name.Equals(propertyName, comparisonType))?.Value;
```

Comment:
Please add parameter null validation. Message can be null. In case the user properties are null I suggest to return null as well.

Reply:
Do you means that?
```
message?.UserProperties?.SingleOrDefault...
```

Reply:
Yes and no 😄 Please throw argument exception if message is null (in both new methods) because this is never allowed. But the user properties may be null when MQTT 3.1.1 is used etc. So here we return null.